Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -1430,8 +1431,14 @@ bool AccessibilityRenderObject::computeIsIgnored() const
 
     // Don't ignore generic focusable elements like <div tabindex=0>
     // unless they're completely empty, with no children.
-    if (isGenericFocusableElement() && node->firstChild())
+    if (isGenericFocusableElement() && node->firstChild()) {
+        RefPtr shadowRoot = node->containingShadowRoot();
+        if (shadowRoot && shadowRoot->mode() == ShadowRootMode::UserAgent && is<HTMLInputElement>(shadowRoot->host())) {
```

Comment:
Is HTMLInputElement the only one that might have this issue? How about textarea, select, <video controls>, etc? I wonder if there's a more general check than casting to HTMLInputElement...

Reply:
Excellent question! I checked textarea, select, audio, and video, and all are fine. So I think just checking for HTMLInputElement is enough!